### PR TITLE
Migrate bourbon and neat from bower to npm

### PR DESCRIPTION
**Pull request by hussam-i-am**
Removes bourbon and bourbon-neat from as bower dependencies, including them instead as node dependencies. I was getting build errors when trying to use the exact versions used in bower, so upgraded them to use the latest versions and had to make some adjustments to the sass. Tested the frontend and compared it to production, and did not come across any major discrepancies, other than some very minor margin changes with the `Back` and `Next` buttons on some of the pages.

### CODE REVIEW

**Comment by hussam-i-am** [comment]
hmmm. guess we'll need to upgrade the major versions if we want the `node-next-test` CI tests to pass, since those versions still depend on `node-sass`

File: frontend/.nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v12.22.6
+v14.21.3
```
Comment (hussam-i-am) [inline frontend/.nvmrc:1]: Updated to match the latest version being used.

File: frontend/app.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "ACME_KEY_1": {
       "description": "used to identify folder with heroku-buildpack-monorepo",
       "value": "frontend"
-    },
+    }
```
Comment (hussam-i-am) [inline frontend/app.json:14]: Fixing some invalid json formatting I noticed

File: frontend/app/styles/bitters/_grid-settings.scss [hunk 3]
```diff
@@ -9,6 +7,3 @@
 // Neat Breakpoints
 $medium-screen: 600px;
 $large-screen: 900px;
-
-$medium-screen-up: new-breakpoint(min-width $medium-screen 4);
-$large-screen-up: new-breakpoint(min-width $large-screen 8);
```
Comment (hussam-i-am) [inline frontend/app/styles/bitters/_grid-settings.scss:14]: Unused so removed these

File: frontend/app/styles/mixins.scss [hunk 4]
```diff
@@ -14,3 +14,22 @@
...
+  *,::after,::before {
+    box-sizing: border-box;
+}
+}
+
+@mixin media($query) {
+  @media screen and (min-width: $query) {
+    @content;
+  }
+}
```
Comment (hussam-i-am) [inline frontend/app/styles/mixins.scss:35]: Added these mixins that were removed in the newer versions of neat

**Review by Tonkpils — APPROVED** [review]

**Pull request merged**